hazelcast/src/jepsen/hazelcast.clj

```diff
@@ -374,29 +813,110 @@
   this is a function, instead of a constant--we may need a fresh workload if we
   run more than one test."
   []
-  {:crdt-map         (map-workload {:crdt? true})
-   :map              (map-workload {:crdt? false})
-   :lock             {:client    (lock-client)
-                      :generator (->> [{:type :invoke, :f :acquire}
-                                       {:type :invoke, :f :release}]
-                                      cycle
-                                      gen/seq
-                                      gen/each)
-                      :checker   (checker/linearizable)
-                      :model     (model/mutex)}
-   :queue            (assoc (queue-client-and-gens)
-                            :checker (checker/total-queue))
-   :atomic-ref-ids   {:client (atomic-ref-id-client nil nil)
-                      :generator (->> {:type :invoke, :f :generate}
-                                      (gen/stagger 1))
-                      :checker  (checker/unique-ids)}
-   :atomic-long-ids  {:client (atomic-long-id-client nil nil)
-                      :generator (->> {:type :invoke, :f :generate}
-                                      (gen/stagger 1))
-                      :checker  (checker/unique-ids)}
-   :id-gen-ids       {:client    (id-gen-id-client nil nil)
-                      :generator {:type :invoke, :f :generate}
-                      :checker   (checker/unique-ids)}})
+  {:crdt-map                     (map-workload {:crdt? true})
+   :map                          (map-workload {:crdt? false})
+   :lock                         {:client    (lock-client "jepsen.lock")
+                                  :generator (->> [{:type :invoke, :f :acquire}
+                                                   {:type :invoke, :f :release}]
+                                                  cycle
+                                                  gen/seq
+                                                  gen/each
+                                                  (gen/stagger 1/10))
+                                  :checker   (checker/linearizable)
+                                  :model     (model/mutex)}
+   :lock-no-quorum               {:client    (lock-client "jepsen.lock.no-quorum")
+                                  :generator (->> [{:type :invoke, :f :acquire}
+                                                   {:type :invoke, :f :release}]
+                                                  cycle
+                                                  gen/seq
+                                                  gen/each
+                                                  (gen/stagger 1/10))
+                                  :checker   (checker/linearizable)
+                                  :model     (model/mutex)}
+   :non-reentrant-cp-lock        {:client    (fenced-lock-client "jepsen.cpLock1")
+                                  :generator (->> [{:type :invoke, :f :acquire :value (.toString (UUID/randomUUID))}
```

Comment:
Did you mean for every acquire to use the same value? Why choose a random value at all?